book/src/smart-contracts-architecture/utils-module.md

```diff
@@ -4,24 +4,42 @@ The Utils module is a various collection of utility functions and tools designed
 
-- [bits.cairo](https://github.com/keep-starknet-strange/satoru/blob/main/src/utils/keys.cairo): Bits constants.
+- [basic_multicall.cairo](https://github.com/keep-starknet-strange/satoru/blob/main/src/utils/basic_multicall.cairo): This utility is tailored to manage and execute batches of function calls on a contract, allowing multiple operations to be performed in a single transaction. It is crucial for optimizing gas utilization and enhancing the efficiency of smart contract interactions. The `multicall` function within this file receives an array of function calls and executes them sequentially. It asserts that the length of the array of calls is greater than zero, ensuring there is data to process, and then iterates over each call, executing them and appending the results to an array. This function is essential for performing multiple operations efficiently and is instrumental in maintaining the robustness and responsiveness of the contract in various scenarios.
 
-- [calc.cairo](https://github.com/keep-starknet-strange/satoru/blob/main/src/utils/keys.cairo): Various calculations.
+- [bits.cairo](https://github.com/keep-starknet-strange/satoru/blob/main/src/utils/bits.cairo): This file contains constants representing bitmasks for different binary word sizes. These constants, `BITMASK_8`, `BITMASK_16`, `BITMASK_32`, and `BITMASK_64`, are designed to facilitate bit-level operations such as bit shifting, bitwise AND, OR, and XOR on `u128` type values within the contract's code. These bitmasks are crucial for performing precise, low-level manipulations of binary data, enabling the effective handling of binary information and the implementation of optimized, robust solutions.
 
-- [enumerable_values.cairo](https://github.com/keep-starknet-strange/satoru/blob/main/src/utils/keys.cairo): Extends EnumerableSet.
+- [calc.cairo](https://github.com/keep-starknet-strange/satoru/blob/main/src/utils/calc.cairo): This file provides a suite of utility functions designed to perform various mathematical calculations and type conversions. It contains functions for rounded-up division, summing numbers with specific return types, calculating the absolute difference between two numbers, and bounded addition and subtraction to prevent overflow and underflow. It also features functions to convert between signed and unsigned integers, ensuring precise and safe calculations are made throughout the contract's code. These utility functions are crucial for maintaining the accuracy and reliability of the mathematical operations performed within the protocol.
```

Comment:
```suggestion
- [calc.cairo](https://github.com/keep-starknet-strange/satoru/blob/main/src/utils/calc.cairo): This file provides a suite of utility functions designed to perform various mathematical calculations and type conversions. It contains functions for rounded-up division, summing numbers with specific return types, calculating the absolute difference between two numbers, and bounded addition and subtraction to prevent overflow and underflow. It also features functions to convert between signed and unsigned integers, ensuring precise and safe calculations are made throughout the contract's code.
```